engines/immortal/immortal.h

```diff
@@ -0,0 +1,783 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef IMMORTAL_IMMORTAL_H
+#define IMMORTAL_IMMORTAL_H
+
+// Audio is only handled in kernal, therefore it is only needed here
+#include "audio/mixer.h"
+
+// Immortal.h is the engine, so it needs the engine headers
+#include "engines/engine.h"
+#include "engines/savestate.h"
+
+// Theorectically, all graphics should be handled through driver, which is part of kernal, which is in immortal.h
+#include "graphics/screen.h"
+#include "graphics/palette.h"
+#include "graphics/surface.h"
+
+// Detection is only needed by the main engine
+#include "immortal/detection.h"
+
+// Disk is only used by immortal.cpp
+#include "immortal/disk.h"
+
+// Common is needed by immortal.h, room.h, and monster.h
+#include "common/debug.h"
```

Comment:
Try to avoid including too many files in other includes, so the header dependency is reduced. Often having just a forward declaration of a class or struct is enough for the header to compile.